src/sql/workbench/parts/notebook/notebook.component.ts

```diff
@@ -441,22 +440,46 @@ export class NotebookComponent extends AngularDisposable implements OnInit, OnDe
 	}
 
 	protected initNavSection(): void {
-		this.addButton(localize('previousButtonLabel', "Previous"),
-			() => this.previousPage());
-		this.addButton(localize('nextButtonLabel', "Next"),
-			() => this.nextPage());
+		this._navProvider = this.notebookService.getNavigationProvider(this._notebookParams.notebookUri);
+
+		if (this.contextKeyService.getContextKeyValue('isDevelopment') &&
+			this.contextKeyService.getContextKeyValue('bookOpened') &&
+			this._navProvider) {
+			this.addButton(localize('previousButtonLabel', "< Previous"),
+				() => this.previousPage());
+			this.addButton(localize('nextButtonLabel', "Next >"),
+				() => this.nextPage());
+
+			this._navProvider.getNavigation(this._notebookParams.notebookUri).then(result => {
+				this.navigationResult = result;
+				this.detectChanges();
+			}, err => {
+				console.log(err);
+			});
+		}
+	}
+
+	public get navigationVisibility(): 'hidden' | 'visible' {
+		if (this.navigationResult) {
+			return this.navigationResult.hasNavigation ? 'visible' : 'hidden';
+		}
+		return 'hidden';
 	}
 
 	private addButton(label: string, onDidClick?: () => void): void {
 		const container = DOM.append(this.bookNav.nativeElement, DOM.$('.dialog-message-button'));
-		let button = new Button(container);
+		let button = new Button(container, {
```

Comment:
Is this accessibility-compliant? Is this going to work for high contrast mode for example?

Reply:
👍 We'd recommend you use CSS rule that a button under the `.book-nav` class should have a certain set of colors. Ideally you should also reference existing theme-compliant color variables for this. Let's chat when you're in to resolve this